hubblestack/extmods/modules/audit.py

```diff
@@ -1,450 +1,556 @@
-Hubble.
-
-Audit checks target a specific audit module, such as ``grep`` or ``fdg``. They
-are formed via YAML files in your hubblestack_data source:
-
-.. code-block:: yaml
-
-    CIS-6.2.4:
-      grep.grep:
-        args:
-          - /etc/group
-        kwargs:
-          pattern: '^+:'
-          fail_on_match: True
-        description: Ensure no legacy "+" entries exist in /etc/group
-
-All audit checks have a few things in common. They have an ``id``, a ``tag``, a
-``description``, and a ``module.function`` (usually with arguments). The ``id`` will be used as the
-tag if a tag is not explicitly provided.
-
-.. code-block:: yaml
-
-    <id>
-      <module.function>:
-        args:
-          - arg1
-          - arg2
-        kwargs:
-          foo: bar
-        description: <description>
-        version: <version>
-        tag: <tag>  # Uses <id> if not defined
-        target: <target>
-        labels:
-          - <label>
-
-You may have noticed there are a few more features shown in the example above.
-Here's how they work:
-
-target:
-    Allows you to use a Salt-style compound match to target this check to
-    specific hosts. If a host doesn't match the target, it won't execute the
-    check.
-
-labels:
-    Allows labels to be applied to a check. Labels will be reported in the
-    results, and can also be targeted by audit runs so only checks with a given
-    label will be executed.
-
-version:
-    Allows checks to be limited to certain versions of hubble. Version
-    requirements can be of the following forms:
-
-    <3.0.0
-    <=3.0.0
-    >3.0.0
-    >=3.0.0
-
-    Multiple version requirements can be used, separated by commas and
-    semicolons. Commas will be processed first, and will result in AND logic.
-    Semicolons will then be processed, using OR logic to combine any existing
-    results. So, to have a check only run on version 3.0 and 3.2 and later
-    (but not run on 3.1) you might do something like this:
-
-    version: '>=3.0.0,<3.1.0;>=3.2.0'
-
-    Note that all checks use distutils.StrictVersion, so this will not work with
-    non-standard hubble releases.
-
-    Checks skipped by version checks will be returned in a separate 'Skipped'
-    key, so that you can track when hosts are skipping targeted checks because
-    they haven't updated Hubble
-
-Like many pieces of Hubble, you can utilize topfiles to target files with one
-or more audit check to specific sets of hosts. Targeting is done via Salt-style
-compound matching:
-
-.. code-block:: yaml
 
-    audit:
-      '*':
-        - cis.linux
-        - adobebaseline
-      'G@os:CoreOS':
-        - coreos_additional_checks
-
-Audit modules take arbitrary args and kwargs and return a tuple
-``(success, data_dict)`` where ``success`` is a boolean True/False on whether
-it was a success or a failure, and ``data_dict`` is a dictionary of any
-information that should be added to the check's data dictionary in the return.
-"""
```

Comment:
Can we keep documentation like this in the file? This specific documentation might not work exactly; can you write something like this and add it to your audit.py?

Reply:
Hi @mew1033,
Thanks for input. I have added documentation for the audit module.